packages/tuono-router/src/components/RouteMatch.spec.tsx

```diff
@@ -7,33 +7,39 @@ import { useServerPayloadData } from '../hooks/useServerPayloadData'
 
 import { RouteMatch } from './RouteMatch'
 
-function createRouteComponent(
-  routeType: string,
-  includeChildren: boolean,
-): RouteComponent {
+function createRouteComponent(routeType: string): RouteComponent {
   const RootComponent = (({ children }: RouteProps) => (
     <div data-testid={routeType}>
       {`${routeType} route`}
-      {includeChildren ? children : null}
+      {children}
     </div>
   )) as RouteComponent
   RootComponent.preload = vi.fn()
   RootComponent.displayName = routeType
   return RootComponent
 }
 
+function createLeafRouteComponent(routeType: string): RouteComponent {
```

Comment:
From @jacobhq review comment https://github.com/tuono-labs/tuono/pull/646#pullrequestreview-2686349679

> Looks good, although I don't understand why we need the createLeafRouteComponent function, would you mind explaining?

Reply:
I think it makes sense having them separate, since passing data creates a weird HTML output that is weird dealing with.